src/dpl/include/dpl/PlacementDRC.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+#include <set>
+#include <string>
+#include <unordered_map>
+
+#include "Coordinates.h"
+
+namespace odb {
+class dbTech;
+class dbOrientType;
+}  // namespace odb
+
+namespace dpl {
+class Grid;
+class Node;
+
+struct EdgeSpacingEntry
+{
+  EdgeSpacingEntry(const int spc_in,
+                   const bool is_exact_in,
+                   const bool except_abutted_in)
+      : spc(spc_in), is_exact(is_exact_in), except_abutted(except_abutted_in)
+  {
+  }
+  bool operator<(const EdgeSpacingEntry& rhs) const { return spc < rhs.spc; }
+  int spc;
+  bool is_exact;
+  bool except_abutted;
```

Comment:
can these be const?

Reply:
I don't think, because of the way I initialize `edge_spacing_table_` in `PlacementDRC::makeCellEdgeSpacingTable`